### DISCUSSION TITLE
Cannot initialize unidirectional M:N relations using loadItems in mongo

### DISCUSSION THREAD

**Discussion by janb87** [mikro-orm/mikro-orm#6234] [General]
### Describe the bug

In our database we have some records that have no property set for a many to many relationship in case there are no items.  This seemed to be the behaviour before mikro orm 6.3.8, I noticed after this upgrade it would always set a value (eg an empty array instead of nothing) 

Eg take following example data

With books:
```ts
{
    name: 'Arthur C. Clark',
    books: [
      { title: "Childhood's End" },
      { title: '2001: A Space Odyssey' },
      { title: 'Rendezvous with Rama' },
    ],
  }
```

Without books:
```ts
{
    name: 'No books',
}
```

Starting from mikro orm version 6.3.8 we started to get issues on loading the collection with no books. It would throw an 'Collection of entity not initialized' error in case the books property was not present in the data.

Following change caused this exception to popup:
https://github.com/mikro-orm/mikro-orm/commit/e53a7d937d50309bafb9113681ca0d7afdbe9485#diff-ed8b851ac26dff47e1a9713dc21c473d50409aa0a38f0185728cf3b34a5bcf7aL207

Would it be possible to have some way to return the old behaviour? I see it was added because of partial loading. Could it for example look at the fields requested in the query and in case the collection is in this list return an empty array instead of undefined.

Another option would be to on our side migrate all the data and add an empty array for documents which are missing the books property.

This item relates to #6079

### Reproduction

It cannot be reproduced using current mongodb version, it only happens if you had data which was created using an older version of mikro orm and you upgraded to a newer version. See scenario described above.

I did create a test which can reproduce the issue on a similar way:
See https://github.com/janb87/mikro-orm/blob/master/tests/features/partial-loading/m-n-owners.mongo.test.ts

### What driver are you using?

@mikro-orm/mongodb

### MikroORM version

6.4.0

### Node.js version

22

### Operating system

Linux

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
> This seemed to be the behaviour before mikro orm 6.3.8

This was always a bug, it wasn't the general behavior I believe. I will take a closer look and try to think about a way out for you, but I don't consider this as a bug in v6.4.

Note that you could always just correct your data, if the value is not there, add an empty array. One way to do this without touching your database could be some entity hook like `@OnLoad`.

> I did create a test which can reproduce the issue on a similar way:

The test behaves the same in v6.3. If you check the queries, you can see that your `No books` entity properly created an empty array in the database:

```
[query] db.getCollection('author').insertMany([ { name: 'Arthur C. Clark', books: [ ObjectId('673461cb2163f47b01fe7264'), ObjectId('673461cb2163f47b01fe7265'), ObjectId('673461cb2163f47b01fe7266') ] }, { name: 'No Books', books: [] } ], {}); [took 3 ms]
```

edit: right, but not in 6.3.7, I see what you mean. still, that was always a bug. will try to think a bit more about this...

**Reply by B4nan** [reply]
We might need a compatibility flag for this, I was hoping the thing that was fixed in 6.3.8 was a regression, but I see it was like that in v5 too.

Either way, the current behavior will be restored in v7, since that is how it was supposed to be working from the very beginning.

**Comment by janb87** [comment]
@B4nan thank you for your time and feedback. Using the '@OnLoad' is a solution which can fix the issue on our side without migrating all our data.
